src/NuGet.Clients/NuGet.VisualStudio.Common/MessageHelper.cs

```diff
@@ -79,6 +79,11 @@ public static void ShowError(ErrorListProvider errorListProvider, TaskErrorCateg
             errorListProvider.ForceShowErrors();
         }
 
+        private static IServiceProvider GetServiceProvider()
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(() => ServiceLocator.GetServiceProviderAsync());
```

Comment:
Is it possible to cache this value?

Reply:
Not easily, all these methods are static. 
Fortunately they are only invoked in rare failure scenarios, so the perf hit is lesser than the effort it'd take to refactor out how this error showing works.